src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/dto/BambooBuildResultNotificationDTO.java

```diff
@@ -111,6 +113,24 @@ public List<BuildLogEntry> extractBuildLogs(ProgrammingLanguage programmingLangu
         return buildLogEntries;
     }
 
+    @JsonIgnore
+    @Override
+    public List<BuildJobDTOInterface> getBuildJobs() {
+        return new ArrayList<>(getBuild().jobs());
```

Comment:
Why do you wrap this in a new ArrayList?

Reply:
`getBuild().jobs()` returns a `List<BambooJobDTO>`. Here we need to return `List<BuildJobDTOInterface>`.
Wrapping it in a new ArrayList is probably the most convenient way to cast the list.

Reply:
Maybe simply define "List<? extends BuildJobDTOInterface>" as return type of the method. (But I'm not sure whether that's ok according to your guidelines :) )